### ISSUE TITLE
[apex] OperationWithLimitsInLoopRule missing other governor-limit consuming operations

### ISSUE THREAD

**Issue by jonathanwiesel** [pmd/pmd#3186]
**Affects PMD Version:** 
6.33

**Description:**

Currently the `OperationWithLimitsInLoopRule` rule verifies for some operations that count against [Apex Governor Limits](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_gov_limits.htm), but some of them are missing and therefore are not reported:

* [`Http.send()`](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_restful_http_http.htm#apex_classes_restful_http_http)
* [`Approval.process()`, `Approval.lock()`, `Approval.unlock()`](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_approval.htm#apex_methods_system_approval)
* [`Messaging.sendEmail()`, `Messaging.renderEmailTemplate()`, `Messaging.renderStoredEmailTemplate()`](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_email_outbound_messaging.htm#apex_classes_email_outbound_messaging)
* [`SObject.recalculateFormulas()`](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_sobject.htm#apex_methods_system_sobject)
* [`System.enqueueJob()`, `System.runAs()`, `System.schedule()`, `System.scheduleBatch()`](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_system.htm#apex_methods_system_system)
* [`Messaging.PushNotification.send()`](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_push_notification.htm)
* [ConnectApi operations](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_connect_api_limits.htm)

**Code Sample demonstrating the issue:**

```
/**
 * @description X
 */
public with sharing class LoopLimits {
    
    /**
     * @description X
     */
    public static void checkLimits() {

        Http h = new Http();
        Approval.ProcessSubmitRequest req = new Approval.ProcessSubmitRequest();
        Messaging.SingleEmailMessage email = new Messaging.SingleEmailMessage();
        Messaging.PushNotification msg = new Messaging.PushNotification();

        for (Integer i = 0; i < 10; i++) {

            h.send(new HttpRequest());

            Eventbus.publish(new Account__e()); //check only required for publish immediately events

            Approval.process(req);
            Approval.lock(new Account());
            Approval.unlock(new Account());

            Messaging.sendEmail(new Messaging.SingleEmailMessage[]{email});
            Messaging.renderEmailTemplate(null, null, null);
            Messaging.renderStoredEmailTemplate(null, null, null);

            new Contact().recalculateFormulas();
            
            System.enqueueJob(new MyQueueable());
            System.runAs(new User()) {
                System.debug(LoggingLevel.DEBUG, 'x');
            }
            System.schedule('x', '0 0 0 1 1 ?', new MySchedule());
            System.scheduleBatch(new MyBatch(), 'x', 1);

            msg.send('x', null);

            ConnectApi.Comment comment = ConnectApi.ChatterFeeds.getComment(null, null);
        }
    }
}
```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1.  Execute PMD over the above snippet, no operation inside the for-loop is reported although they all consume governor limits
